src/extlibs/hrvo/hrvo_test.cpp

```diff
@@ -0,0 +1,517 @@
+    World world = World(Field::createSSLDivisionBField(),
+                        Ball(Point(0, 0), Vector(0, 0), Timestamp()),
+                        Team({friendly_robot}), Team());
+
+    sim.updateWorld(world);
+
+    EXPECT_EQ(1, sim.getNumAgents());
+
+    auto agents = sim.getAgents();
+    assertRobotInAgentList(friendly_robot, AgentType::FRIENDLY, sim.getAgents());
+}
+
+TEST_F(TestHrvo, test_update_world_with_one_enemy_agent)
+{
+    Robot enemy_robot =
+        Robot(0, Point(), Vector(), Angle(), AngularVelocity(), Timestamp());
+    HRVOSimulator sim =
+        HRVOSimulator(200u, create2021RobotConstants(), TeamColour::YELLOW);
+    World world =
+        World(Field::createSSLDivisionBField(),
+              Ball(Point(0, 0), Vector(0, 0), Timestamp()), Team(), Team({enemy_robot}));
+
+    sim.updateWorld(world);
+
+    ASSERT_EQ(1, sim.getNumAgents());
+
+    auto agents = sim.getAgents();
+    assertRobotInAgentList(enemy_robot, AgentType::ENEMY, sim.getAgents());
+}
+
+TEST_F(TestHrvo, test_update_world_with_friendly_and_enemy_agent)
+{
+    Robot friendly_robot =
+        Robot(1, Point(1, 1), Vector(1, 1), Angle(), AngularVelocity(), Timestamp());
+    Robot enemy_robot =
+        Robot(0, Point(), Vector(), Angle(), AngularVelocity(), Timestamp());
+    HRVOSimulator sim =
+        HRVOSimulator(200u, create2021RobotConstants(), TeamColour::YELLOW);
+    World world = World(Field::createSSLDivisionBField(),
+                        Ball(Point(0, 0), Vector(0, 0), Timestamp()),
+                        Team({friendly_robot}), Team({enemy_robot}));
+
+    sim.updateWorld(world);
+
+    ASSERT_EQ(2, sim.getNumAgents());
+
+    auto agents = sim.getAgents();
+    assertRobotInAgentList(friendly_robot, AgentType::FRIENDLY, sim.getAgents());
+    assertRobotInAgentList(enemy_robot, AgentType::ENEMY, sim.getAgents());
+}
+
+TEST_F(TestHrvo, test_update_world_add_friendly_robots_second_tick)
+{
+    Robot friendly_robot_1 =
+        Robot(0, Point(), Vector(), Angle(), AngularVelocity(), Timestamp());
+    Robot friendly_robot_2 =
+        Robot(1, Point(1, 1), Vector(), Angle(), AngularVelocity(), Timestamp());
+    Robot friendly_robot_3             = Robot(2, Point(), Vector(1, 1), Angle::half(),
+                                   AngularVelocity(), Timestamp::fromSeconds(1));
+    std::vector<Robot> friendly_robots = {friendly_robot_1, friendly_robot_2,
+                                          friendly_robot_3};
+
+    Robot enemy_robot_1 =
+        Robot(1, Point(-1, -1), Vector(), Angle(), AngularVelocity(), Timestamp());
+    std::vector<Robot> enemy_robots = {enemy_robot_1};
+
+    HRVOSimulator sim =
+        HRVOSimulator(200u, create2021RobotConstants(), TeamColour::YELLOW);
+    World world =
+        World(Field::createSSLDivisionBField(), Ball(Point(), Vector(), Timestamp()),
+              Team(friendly_robots), Team(enemy_robots));
+    sim.updateWorld(world);
```

Comment:
If you want to be adding robots on the second tick (which is a good idea as a way to have the KD-Tree updating in the test) you should also call `sim.doStep()`. Similarly for the tests below

Reply:
done